### PR TITLE
Match timer

### PR DESCRIPTION
**Pull request by lauracrodriguezv**
- I updated the time left depending on current match state replicating the server time on the game state, to be updated on the HUD

### CODE REVIEW

File: Source/BestProjectEver/Public/Core/GameModes/BPE_GameplayGameMode.h [hunk 1]
```diff
@@ -17,23 +18,53 @@ class BESTPROJECTEVER_API ABPE_GameplayGameMode : public AGameMode
...
+	UPROPERTY(EditDefaultsOnly, Category="Game Mode")
+	float MatchTime;
+	
+	UPROPERTY(EditDefaultsOnly, Category="Game Mode")
+	float WarmupTime;
+
+	float LevelStartingTime;
+
+	/** rate to update time left */
+	UPROPERTY(Transient)
```
Comment (JDAlzate) [inline Source/BestProjectEver/Public/Core/GameModes/BPE_GameplayGameMode.h:35]: Timer handles have no need of being UPROPERTYs

**Review by JDAlzate — APPROVED** [review]
Approved with the change we discussed :) The other is just a small comment, not crucial in any way

File: Source/BestProjectEver/Private/HUD/BPE_HUD.cpp [hunk 2]
```diff
@@ -5,13 +5,23 @@
...
 #include "HUD/Widgets/BPE_CharacterOverlay.h"
 #include "GameFramework/PlayerController.h"
+#include "HUD/Widgets/BPE_TimerWidget.h"
 
 //----------------------------------------------------------------------------------------------------------------------
 void ABPE_HUD::BeginPlay()
 {
 	Super::BeginPlay();
+	InitializeReferences();
+	GetWorldTimerManager().SetTimer(TimerHandle_RefreshTimeLeftRate, this, &ABPE_HUD::UpdateMatchTimer, 1.0f, true, 0.0f);
```
[the remark this answers is not in the document]
Comment (lauracrodriguezv) [reply Source/BestProjectEver/Private/HUD/BPE_HUD.cpp]: image
image
Done <3

File: Source/BestProjectEver/Public/Core/GameModes/BPE_GameplayGameMode.h (hunk 1 above)
Comment (lauracrodriguezv) [reply Source/BestProjectEver/Public/Core/GameModes/BPE_GameplayGameMode.h:39]: Thanksss I erase it and replace it with a delegate

**Comment by mcdrakteravision** [comment]
merged but 👌

**Pull request merged**